tests/src/FunctionalJavascript/ContributionPageTest.php

```diff
@@ -120,27 +89,21 @@ public function testSubmitContribution() {
     $this->getSession()->getPage()->fillField('Street Address', '123 Milwaukee Ave');
     $this->getSession()->getPage()->fillField('City', 'Milwaukee');
 
-    // Select2 is being difficult; unhide the country select.
+    // Select2 is being difficult; unhide the country and state/province select.
     $driver = $this->getSession()->getDriver();
     assert($driver instanceof DrupalSelenium2Driver);
     $driver->executeScript("document.getElementById('billing_country_id-5').style.display = 'block';");
-    $this->getSession()->getPage()->fillField('billing_country_id-5', '1228');
+    $driver->executeScript("document.getElementById('billing_state_province_id-5').style.display = 'block';");
 
-    // @todo find a way to better wait for state/pronvince to populate.
+    $this->getSession()->getPage()->fillField('billing_country_id-5', '1228');
+    // Wait for select2's AJAX request.
     $this->assertSession()->assertWaitOnAjaxRequest();
-    sleep(1);
-
-    // Select2 is being difficult; unhide the state/province. select.
-    $driver->executeScript("document.getElementById('billing_state_province_id-5').style.display = 'block';");
+    $this->getSession()->wait(1000, 'document.getElementById("billing_state_province_id-5").options.length > 1');
```

Comment:
This speeds up the contribution page test a bit by being smarter for when we can set the state/province value.